### PR TITLE
Fix SamlAuthenticationIT flakyness

### PR DESCRIPTION
**Pull request by breskeby**

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java [hunk 1]
```diff
@@ -181,7 +185,7 @@ public void execute() {
             }
 
             final List<String> tags = parameters.getTags().get();
-            final boolean isCrossPlatform = parameters.getPlatform().get().equals(Architecture.current().dockerPlatform) == false;
+            final boolean isCrossPlatform = isCrossPlatform();
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java:188]: we tweaked this to allow reusing this task and supporting multi platform images

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java [hunk 2]
```diff
@@ -203,18 +207,34 @@ public void execute() {
                 tags.forEach(tag -> spec.args("--tag", tag));
 
                 parameters.getBuildArgs().get().forEach((k, v) -> spec.args("--build-arg", k + "=" + v));
+
+                if (parameters.getPush().getOrElse(false)) {
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java:211]: allow pushing directly from here. Makes it easier to push multiple images within a gradle build without the need of doing as part of a script after the build as we use to do for distribution images

**Comment by breskeby** [comment]
This looks stable now: https://buildkite.com/elastic/elasticsearch-pull-request/builds/8265

File: .buildkite/scripts/fixture-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -euo pipefail
+
+echo "$DOCKER_REGISTRY_PASSWORD" | docker login -u "$DOCKER_REGISTRY_USERNAME" --password-stdin docker.elastic.co
+unset DOCKER_REGISTRY_USERNAME DOCKER_REGISTRY_PASSWORD
+
+docker buildx create --use
+.ci/scripts/run-gradle.sh deployFixtureDockerImages
```
Comment (breskeby) [inline .buildkite/scripts/fixture-deploy.sh:9]: prep for setting up a ci pipeline for building and deploying docker based fixtures once per day https://github.com/elastic/elasticsearch-infra/pull/378

File: x-pack/test/idp-fixture/src/main/resources/idp/shibboleth-idp/conf/admin/metrics.xml [hunk 4]
```diff
@@ -9,42 +9,14 @@
 
     default-init-method="initialize"
     default-destroy-method="destroy">
-
```
Comment (breskeby) [inline x-pack/test/idp-fixture/src/main/resources/idp/shibboleth-idp/conf/admin/metrics.xml:12]: While debugging I noticed some unused metrics are setup. removed more output noise than startup time though

File: x-pack/test/idp-fixture/build.gradle [hunk 5]
```diff
@@ -7,3 +11,26 @@ dependencies {
...
+  baseImages = ["osixia/openldap:1.4.0"]
+  noCache = BuildParams.isCi
+  tags = ["docker.elastic.co/elasticsearch-dev/openldap-fixture:1.0"]
+  getPush().set(true)
+  getPlatforms().addAll( Architecture.values().collect{ it.dockerPlatform } )
+}
+
+tasks.register("deployFixtureDockerImages") {
+  dependsOn tasks.withType(DockerBuildTask)
+}
```
Comment (breskeby) [inline x-pack/test/idp-fixture/build.gradle:36]: We can move parts of this to an internal plugin once we do this fixture pre baking for more fixtures. We can evaluate existing fixtures in a later pr

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java (hunk 1 above)
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java:188]: Ah, neat. I didn't know you could pass multiple platforms to a single task.

Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java:188]: Seems unnecessary to have this field now that this logic lives in the `isCrosPlatform()` method.

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java [hunk 6]
```diff
@@ -55,15 +58,13 @@ public class DockerBuildTask extends DefaultTask {
...
     private MapProperty<String, String> buildArgs;
-    private Property<String> platform;
 
     @Inject
     public DockerBuildTask(WorkerExecutor workerExecutor, ObjectFactory objectFactory, ProjectLayout projectLayout) {
         this.workerExecutor = workerExecutor;
         this.markerFile = objectFactory.fileProperty();
         this.dockerContext = objectFactory.directoryProperty();
         this.buildArgs = objectFactory.mapProperty(String.class, String.class);
-        this.platform = objectFactory.property(String.class).convention(Architecture.current().dockerPlatform);
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java:66]: FYI, this change is going to have implications for the serverless build, we'll want to follow up with a fix there once this is merged.

File: x-pack/test/idp-fixture/build.gradle (hunk 5 above)
Comment (mark-vieira) [reply x-pack/test/idp-fixture/build.gradle:36]: Sounds good.

File: x-pack/test/idp-fixture/build.gradle [hunk 7]
```diff
@@ -7,3 +11,26 @@ dependencies {
   api project(':test:fixtures:testcontainer-utils')
   api "junit:junit:${versions.junit}"
 }
+
+tasks.register("deployIdpFixtureDockerImages", DockerBuildTask) {
+    dockerContext.fileValue(file("src/main/resources/idp"))
+    baseImages = ["openjdk:11.0.16-jre"]
+    noCache = BuildParams.isCi
+    tags = ["docker.elastic.co/elasticsearch-dev/idp-fixture:1.0"]
+    getPush().set(BuildParams.isCi)
```
Comment (mark-vieira) [inline x-pack/test/idp-fixture/build.gradle:20]: should be able to simplify this as `push = BuildParams.isCI`

File: x-pack/test/idp-fixture/src/main/java/org/elasticsearch/test/fixtures/idp/IdpTestContainer.java [hunk 8]
```diff
@@ -35,107 +33,7 @@ protected IdpTestContainer() {
...
-                    .withStatement(
-                        new SingleArgumentStatement(
-                            "HEALTHCHECK",
-                            "CMD curl -f -s --http0.9 http://localhost:4443 " + "--connect-timeout 10 --max-time 10 --output - > /dev/null"
-                        )
-                    )
-                    // .expose(4443)
-                    .build()
-            )
+            new ImageFromDockerfile("es-idp-testfixture").withDockerfileFromBuilder(builder -> builder.from(DOCKER_BASE_IMAGE).build())
```
Comment (mark-vieira) [inline x-pack/test/idp-fixture/src/main/java/org/elasticsearch/test/fixtures/idp/IdpTestContainer.java:36]: I think we want to use `RemoteDockerImage` here instead. That should avoid building essentially an "empty" Dockerfile that just extends from the remote one.

File: x-pack/test/idp-fixture/src/main/java/org/elasticsearch/test/fixtures/idp/OpenLdapTestContainer.java [hunk 9]
```diff
@@ -30,35 +30,11 @@ public OpenLdapTestContainer() {
...
-                    )
-                    .copy(
-                        "openldap/ldif/config.ldif",
-                        "/container/service/slapd/assets/config/bootstrap/ldif/custom/10-bootstrap-config.ldif"
-                    )
-                    .copy("openldap/certs", "/container/service/slapd/assets/certs")
-
-                    .build()
-            )
+            new ImageFromDockerfile("es-openldap-testfixture").withDockerfileFromBuilder(builder -> builder.from(DOCKER_BASE_IMAGE).build())
```
Comment (mark-vieira) [inline x-pack/test/idp-fixture/src/main/java/org/elasticsearch/test/fixtures/idp/OpenLdapTestContainer.java:33]: Same as before, we should use `RemoteDockerImage`.

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**